### PR TITLE
HDS-1863 remove deprecated

### PR DESCRIPTION
**Pull request by mikkojamG**
## Description

Remove deprecated components for release 3.0.0. Mostly worked on old `Navigation` component, old style based `Tag` and `StatusLabel` deprecations seem to have already been removed.

- Remove old `Navigation` components with its sub-components and replace with new `Header` components in places where previously used (also doc site layout).
- Cookie consent modal, replaced `Navigation` `LanguageSelector` by implementing the solution directly to component

## Related Issue

Closes [HDS-1863](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1863)

## Screenshots (if appropriate):

[HDS-1863]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-1863?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mrTuomoK — COMMENTED** [review]
Overall nice job 👍  I jsut had that one comment I left about using the Header.Link outside the Header itself.

File: packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcher.tsx [hunk 1]
```diff
@@ -19,15 +21,18 @@ export function LanguageSwitcher(): React.ReactElement {
...
       label={currentOption.label}
+      menuOffset={10}
       buttonAriaLabel={languageSelectorAriaLabel}
-      className={styles.languageSelectorOverride}
-      id={languageSelectorId}
+      closeOnItemClick
     >
       {languageOptions.map((option) => (
-        <Navigation.Item
+        <Header.Link
```
[the remark this answers is not in the document]
Comment (mrTuomoK) [reply packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcher.tsx:35]: I don't say it's wrong, just feels weird 🤔

[the remark this answers is not in the document]
Comment (mikkojamG) [reply packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcher.tsx:35]: I guess the alternate solution would be to recreate previous Navigation.Item for the cookie consent 🤔

[the remark this answers is not in the document]
Comment (mikkojamG) [reply packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcher.tsx]: Replicated old Navigation.Item as LanguageSwitcherItem https://github.com/City-of-Helsinki/helsinki-design-system/pull/1115/commits/2d1c11bfca1781c96aa9e620a650e38a92255584

File: packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcherItem/LanguageSwitcherItem.module.scss [hunk 2]
```diff
@@ -93,7 +88,7 @@
     padding: var(--spacing-2-xs);
 
     &:focus {
-      box-shadow: 0 0 0 var(--header-focus-outline-width) var(--item-focus-outline-color);
+      box-shadow: 0 0 0 3px var(--color-coat-of-arms);
```
Comment (mrTuomoK) [inline packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcherItem/LanguageSwitcherItem.module.scss:91]: Do changing from variable names to color variables affect the possibility to theme the components? (If you know what I mean)

Comment (mrTuomoK) [reply packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcherItem/LanguageSwitcherItem.module.scss:91]: Dunno if this is needed for shadows though?

File: packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcherItem/LanguageSwitcherItem.module.scss [hunk 3]
```diff
@@ -158,7 +153,7 @@
   .dropdownItem.active:after,
   .dropdownLinkItem:hover:not(:focus-within):after,
   .dropdownLinkItem.active:after {
-    background-color: var(--item-active-color);
+    background-color: var(--color-black-90);
```
Comment (mrTuomoK) [inline packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcherItem/LanguageSwitcherItem.module.scss:156]: Same theming comment here as previously?

File: packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcherItem/LanguageSwitcherItem.module.scss (hunk 2 above)
Comment (mikkojamG) [reply packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcherItem/LanguageSwitcherItem.module.scss:91]: These are previously inherited from deprecated Navigation component and no longer would have effect here. Thus root level variables 🤔

File: packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcherItem/LanguageSwitcherItem.module.scss (hunk 3 above)
Comment (mikkojamG) [reply packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcherItem/LanguageSwitcherItem.module.scss:156]: These are previously inherited from deprecated Navigation component and no longer would have effect here. Thus root level variables 🤔

File: packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcherItem/LanguageSwitcherItem.module.scss (hunk 2 above)
Comment (mrTuomoK) [reply packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcherItem/LanguageSwitcherItem.module.scss:91]: Ok, maybe (maybe) it then inherit from the CookieConsent but yeah it doesn't at least yer have theming in it 🤷‍♂️

**Review by mrTuomoK — APPROVED** [review]
This one passes OK, I guess we don't quite like it this way but yeah, the whole CookieConsent will be redone sometime later so no use tweaking this one any more than needed. 👍

File: packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcher.tsx [hunk 4]
```diff
@@ -19,15 +21,18 @@ export function LanguageSwitcher(): React.ReactElement {
...
+    <MenuButton
+      className={classNames(styles.languageDropdown)}
+      id={languageSelectorId}
       label={currentOption.label}
+      menuOffset={10}
       buttonAriaLabel={languageSelectorAriaLabel}
-      className={styles.languageSelectorOverride}
-      id={languageSelectorId}
+      closeOnItemClick
     >
```
[the remark this answers is not in the document]
Comment (mikkojamG) [reply packages/react/src/components/cookieConsent/languageSwitcher/LanguageSwitcher.tsx:33]: Good catch, was due to ineffective variable: https://github.com/City-of-Helsinki/helsinki-design-system/pull/1115/commits/8212e0e2dcb09537c8bc7ffec7171b51c857b304

**Pull request merged**